Lib/unittest/mock.py

```diff
@@ -2482,6 +2484,55 @@ def __set__(self, obj, val):
         self(val)
 
 
+class WaitableMock(MagicMock):
+    """
+    A mock that can be used to wait until it was called.
+
+    `event_class` - Class to be used to create event object.
+    Defaults to Threading.Event and can take values like multiprocessing.Event.
+    """
+
+    def __init__(self, *args, event_class=threading.Event, **kwargs):
+        _safe_super(WaitableMock, self).__init__(*args, **kwargs)
+        self._event = event_class()
+        self._expected_calls = {}
```

Comment:
This might need to be a default dict of the event class the user passed. 
Otherwise the following situation might fail:
Some calls ‘wait_untill_called_with’ after te call has been performed (it should not block)

Reply:
Thanks Mario, I have added a test and used `defaultdict` which simplifies the code.